code-challenges.test.js

```diff
@@ -27,8 +27,35 @@ const colors2 = [
 ]
 // Expected output example (can be a different order): ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
 
+// JEST TEST
+// Describe funtion("colorsShuffled"), () => {
+//  it ([returns a shuffled array with the first coler deleted] from colors1)
+// it ([returns a shuffled array with the first color deleted from colors2])
+// expected output: ["yellow", "blue", "pink", "green"]
+// expected output: ["saffron", "aquamarine", "periwinkle", "indigo", "ochre"]
+//})
+
 // b) Create the function that makes the test pass.
 
+// Psuedo-code
+// 1. Create function(colorsShuffled)
+// 2. Using the .shift method, remove the first item from each array
+// 3. Using a for loop, create function var array1 to store the randomized values of the remaining elements of the array
+// 4. Console log the shuffled array
+
+  const colorsShuffled = (array) => {
+  let array1 = array.shift(0)
+  // for (arrayX = 0; i < arrayX.length; i++) {
+  //   var j = Math.floor(Math.random() * (i - 1));
+  //   var temp = array[i];
+  //   array[i] = array[j];
+  //   array[j] = temp;
+  // }
+}
+console.log(colors1)
+
+//  I could not get the .shift method to work in this problem, which kept me from getting the second part of the problem.  I could only get it to reverse the order of the elements of the array without removing the first one before reversal. My intent was to use .shift to remove the first element of the array, and then use a for loop to shuffle the remaining elements.
+ 
```

Comment:
Here is one approach to this prompt.
Jest test
```javascript
describe('colorsShuffled', () => {
  it('takes in an array, removes the first item from the array and shuffles the remaining content', () => {
    // test variables
    const colors1 = ["purple", "blue", "green", "yellow", "pink"]
    const colors2 = ["chartreuse", "indigo", "periwinkle", "ochre", "aquamarine", "saffron"]

    // expect statement using the expect.arrayContaining method to state that the array will contain these values but does not need to match the order as given
    expect(colorsShuffled(colors1)).toEqual(expect.arrayContaining(["blue", "green", "yellow", "pink"]))
    expect(colorsShuffled(colors2)).toEqual(expect.arrayContaining(["indigo", "periwinkle", "ochre", "aquamarine", "saffron"]))
  })
})
```
function with explanation
```javascript
    const colorsShuffled = (array) => {
      // Shift is a mutator. If you store it in a variable, it will store the value that was removed. We want access to the remaining values so just run the method without storing it in a variable. No arguments or parameters are required in the ()
      array.shift()

      // the for loop
      // start the count at 0, using variable `i` to store it
      // use the length to help dynamically tell the for loop where to stop the count
      // the count will increase by 1 on each iteration
      for(i = 0; i < array.length - 1; i++) {

        // swap values at two different indexes within the array
        // store a random number that will represent an index within the array
        var j = Math.floor(Math.random() * i);
        // store the value at the index represented by number produced from for loop
        var temp = array[i];
        // reassign the value at the i-index with the value at the random j-index
        array[i] = array[j];
        // reassign the value at the random index with what was store in the temp variable
        array[j] = temp;
      }
      // need to return outside the for loop to see the final array
      return array
    }
```